cli/scripts/e2e/e2e.ts

```diff
@@ -487,6 +491,43 @@ function runSmokeTests<T>(
     }
   );
 
+  suite(
+    `${npmClient} passes through correct args ${
+      options.cwd ? " from " + options.cwd : ""
+    }`,
+    async () => {
+      const result = getCommandOutputAsArray(
+        repo.turbo(
+          "run",
+          ["args", "--filter=//", "--", "--script-arg=42"],
```

Comment:
Can we do quite a few more variations on this with varying argument patterns? That'll likely make the rest of the stuff here need to become utility functions of some sort.

I'm thinking mostly edge cases:
- `-- -- --`
- ``
- `-f --f ---f ----f`
- `--same -- --same`
- `--zero -- --one -- --two -- --three`
- `-------------------------`

In terms of test understandability, I'd love to be able to spec the input like this and then the output expected adjacent to it. (e.g. Go table tests)

We might should also create multiple scripts in `package.json` that log out `argv` with and without their own separator.

Reply:
> We might should also create multiple scripts in package.json that log out argv with and without their own separator.

Could you elaborate on this?

Reply:
This is really an unnecessary, overly-defensive idea. Basically, we have a script called `argv` in the root `package.json` now that is just an `argv` echo service.

We could create variations on that script that had other `--` setups to detect if _node itself_ changes the way it works.

But that'll realistically break _way_ more than us and I can't see that being something we truly need to test.